iothub_client/inc/iothub_device_client.h

```diff
@@ -349,6 +348,133 @@ extern "C"
+    *           `IoTHubDeviceClient_CreateUploadContext`
+    *           `IoTHubDeviceClient_AzureStoragePutBlock`
+    *           `IoTHubDeviceClient_DestroyUploadContext`
+    *           For the standard less-granular uploads to blob please use either
+    *           `IoTHubDeviceClient_UploadToBlobAsync` or `IoTHubDeviceClient_UploadMultipleBlocksToBlobAsync`.
+    *
+    * @param    uploadContextHandle    The handle created with `IoTHubDeviceClient_CreateUploadContext`.
+    * @param    blockNumber            Number of the block being uploaded.
+    * @param    dataPtr                Pointer to the block data to be uploaded to Azure Storage blob.
+    * @param    dataSize               Size of the block data pointed by `dataPtr`.
+    *
+    * @warning  This is a synchronous/blocking function.
+    *
+    * @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+    */
+    MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubDeviceClient_AzureStoragePutBlock, IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE, uploadContextHandle, uint32_t, blockNumber, const uint8_t*, dataPtr, size_t, dataSize);
+
+    /**
+    * @brief    This API performs an Azure Storage PUT BLOCK LIST operation.
+    * @remark   It is part of a set of functions for more granular control over Azure IoT-based blob uploads.
+    *           This function is expected to be used along with:
+    *           `IoTHubDeviceClient_CreateUploadContext`
+    *           `IoTHubDeviceClient_AzureStoragePutBlock`
+    *           `IoTHubDeviceClient_DestroyUploadContext`
+    *           For the standard less-granular uploads to blob please use either
+    *           `IoTHubDeviceClient_UploadToBlobAsync` or `IoTHubDeviceClient_UploadMultipleBlocksToBlobAsync`.
+    *           If this function fails (due to any HTTP error to Azure Storage) it can
+    *           be run again for a discretionary number of times in an attempt to succeed after, for example,
+    *           an internet connectivity disruption is over.  
+    *
+    * @param    uploadContextHandle    The handle created with `IoTHubDeviceClient_CreateUploadContext`.
+    *
+    * @warning  This is a synchronous/blocking function.
+    *
+    * @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+    */
+    MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubDeviceClient_AzureStoragePutBlockList, IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE, uploadContextHandle);
+
+    /**
+    * @brief    This API destroy a blob upload context previously created with `IoTHubDeviceClient_CreateUploadContext`.
+    * @remark   It is part of a set of functions for more granular control over Azure IoT-based blob uploads.
+    *           This function is expected to be used along with:
+    *           `IoTHubDeviceClient_CreateUploadContext`
+    *           `IoTHubDeviceClient_AzureStoragePutBlock`
+    *           `IoTHubDeviceClient_NotifyUploadCompletion`
```

Comment:
Should this one be in the other comments as well?

Reply:
Yes. Fixed.